src/logjam-util.c

```diff
@@ -619,6 +619,41 @@ zmsg_savex (zmsg_t *self, FILE *file)
     return 0;
 }
 
+// Save the payload frame of the message only
+int zmsg_savex_payload (zmsg_t *self, FILE *file) 
+{
+    assert (self);
+    assert (zmsg_is (self));
+    assert (file);
+
+    zframe_t *frame = zmsg_first (self); //stream frame
+    frame = zmsg_next (self);  //topic frame
+    frame = zmsg_next (self);  // payload frame
+    
+    msg_meta_t meta;
+    msg_extract_meta_info(self, &meta);
+    int compression_method = meta.compression_method;
+    if (compression_method) {
+        char *body;
+        size_t body_len;
+        zchunk_t *buffer = zchunk_new(NULL, INITIAL_DECOMPRESSION_BUFFER_SIZE);
+        int rc = decompress_frame(frame, compression_method, buffer, &body, &body_len);
+        zchunk_destroy(&buffer);
```

Comment:
You can't delete the buffer here, as body will point to the compression buffer. I would allocate the buffer in the main program and pass it in as a parameter to this function.

Also, I would change the name of the function, as it does not really save the message, but the json payload.

So, dump_json_payload or similar seem better to me.